### ISSUE TITLE
prebuilt binaries for electron-v1.8-*

### ISSUE THREAD

**Issue by mark-buer** [grpc/grpc-node#230]
This isn't an issue so much as a feature request.

Prebuilt binaries (for v1.10.0) appear to be available for `electron-v1.7-*`, but not for `electron-v1.8-*`.

Is there any chance that binaries for electron v1.8 could be added?

Reason for asking is that the lack of binaries complicates CI for our electron app (more setup required and builds take *much* longer).

**Comment by nicolasnoble** [comment]
I thought we have added electron 1.8 support for 1.10. That's odd. We'll
look into it.

**Comment by nicolasnoble** [comment]
This is fairly odd. 1.8 was definitely added for 1.10, but doesn't seem to be present upstream indeed.

I've uploaded a 1.10.1-pre1 build which should contain it. Could you give it a go ?

**Comment by mark-buer** [comment]
I'm not sure where I'm supposed to be looking for the 1.10.1-pre1 build. Npm/yarn doesn't recognise that version, and I cannot see any git tag or github release named likewise.

**Comment by nicolasnoble** [comment]
Right. I didn't push the actual npm package, only the node-pre-gyp binaries. Sorry. Long day. I'll update once I've pushed it.

**Comment by ponkin** [comment]
@nicolasnoble any updates on this?

**Comment by nicolasnoble** [comment]
Ah, sorry for not coming back here. During the build, we detected an issue with the packages, and filed a bug with node-pre-gyp: https://github.com/mapbox/node-pre-gyp/issues/356

**Comment by callmecary** [comment]
I happened to find that 'electron-rebuild grpc' throws build errors on my MacBook , could it be related? It works fine on Windows though. Thanks. 
Electron version: 1.8.4, Electron-rebuild version, 1.7.3

**Comment by nicolasnoble** [comment]
The best method for building I think is the following command:
```
HOME=~/.electron-build npm rebuild --runtime=electron --target=1.8.4 --disturl=https://atom.io/download/electron
```

**Comment by callmecary** [comment]
@nicolasnoble thanks for reply. I ran the commands you suggested, but got the following errors at the top:
```node-pre-gyp ERR! Tried to download(403): https://storage.googleapis.com/grpc-precompiled-binaries/node/grpc/v1.10.0/electron-v1.8-darwin-x64-unknown.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for grpc@1.10.0 and electron@1.8.4 (electron-v1.8 ABI, unknown) (falling back to source compile with node-gyp)```

**Comment by nicolasnoble** [comment]
But that's not an error, this should be followed by node-gyp building the
extension...

**Comment by callmecary** [comment]
```Traceback (most recent call last):
  File "./gyp-mac-tool", line 611, in <module>
    sys.exit(main(sys.argv[1:]))
  File "./gyp-mac-tool", line 28, in main
    exit_code = executor.Dispatch(args)
  File "./gyp-mac-tool", line 43, in Dispatch
    return getattr(self, method)(*args[1:])
  File "./gyp-mac-tool", line 246, in ExecFilterLibtool
    if not libtool_re.match(line) and not libtool_re5.match(line):
TypeError: cannot use a string pattern on a bytes-like object
make: *** [Release/libgrpc.a] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Darwin 17.4.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--library=static_library" "--module=/Users/mr_maggie/workspace/chainhub/node_modules/grpc/src/node/extension_binary/electron-v1.8-darwin-x64-unknown/grpc_node.node" "--module_name=grpc_node" "--module_path=/Users/mr_maggie/workspace/chainhub/node_modules/grpc/src/node/extension_binary/electron-v1.8-darwin-x64-unknown"
gyp ERR! cwd /Users/mr_maggie/workspace/chainhub/node_modules/grpc
gyp ERR! node -v v8.9.4
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --library=static_library --module=/Users/mr_maggie/workspace/chainhub/node_modules/grpc/src/node/extension_binary/electron-v1.8-darwin-x64-unknown/grpc_node.node --module_name=grpc_node --module_path=/Users/mr_maggie/workspace/chainhub/node_modules/grpc/src/node/extension_binary/electron-v1.8-darwin-x64-unknown' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/mr_maggie/workspace/chainhub/node_modules/grpc/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:126:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:925:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
node-pre-gyp ERR! System Darwin 17.4.0
node-pre-gyp ERR! command "/usr/local/bin/node" "/Users/mr_maggie/workspace/chainhub/node_modules/grpc/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--library=static_library"
node-pre-gyp ERR! cwd /Users/mr_maggie/workspace/chainhub/node_modules/grpc
node-pre-gyp ERR! node -v v8.9.4
node-pre-gyp ERR! node-pre-gyp -v v0.6.39
node-pre-gyp ERR! not ok
Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --library=static_library --module=/Users/mr_maggie/workspace/chainhub/node_modules/grpc/src/node/extension_binary/electron-v1.8-darwin-x64-unknown/grpc_node.node --module_name=grpc_node --module_path=/Users/mr_maggie/workspace/chainhub/node_modules/grpc/src/node/extension_binary/electron-v1.8-darwin-x64-unknown' (1)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! grpc@1.10.0 install: `node-pre-gyp install --fallback-to-build --library=static_library`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the grpc@1.10.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.```

**Comment by nicolasnoble** [comment]
Sadly, that error isn't specific enough to let us know what went wrong.

Anyhow, we've published 1.10.1 by downgrading node-pre-gyp, and it should contain Electron 1.8 binaries.

**Comment by callmecary** [comment]
Thank you so much, that works!
